src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -1333,6 +1224,94 @@ def construct_payload(self):
         self.set_up_service_binds()
         self.set_up_extended_location()
 
+        if self.get_argument_source():
```

Comment:
Recommend to add `if self.get_argument_source()` in` self._set_up_source_or_repo()`, not in outside.

Reply:
Updated